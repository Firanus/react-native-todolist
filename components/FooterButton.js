import React, { Component } from 'react';
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';

export default class FooterButton extends Component {
  render() {
    return (
      <TouchableOpacity style={styles.outerContainer}
                               onPress={this.props.buttonHandler}>
        <View style={styles.innerContainer}>
          <Text style={styles.text}>{this.props.children}</Text>
        </View>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  innerContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  outerContainer: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    //backgroundColor: '#ff323c',
    borderTopColor: '#f2f2f2',
    borderTopWidth: 1,
    padding: 15,
    height: 70,
  },
  text: {
    color: '#ff323c',
  }
});