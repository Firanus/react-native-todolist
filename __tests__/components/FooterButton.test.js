import React from 'react';
import FooterButton from '../../components/FooterButton';

import renderer from 'react-test-renderer';

it('renders without c', () => {
  const emptyTestFunction = () => {};
  const FooterText = "Test Text";

  const rendered = renderer.create(<FooterButton buttonHandler={emptyTestFunction()}>{FooterText}</FooterButton>).toJSON();
  expect(rendered).toMatchSnapshot();
});
