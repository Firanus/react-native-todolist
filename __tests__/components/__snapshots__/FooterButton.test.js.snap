// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without c 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderTopColor": "#f2f2f2",
      "borderTopWidth": 1,
      "bottom": 0,
      "height": 70,
      "left": 0,
      "opacity": 1,
      "padding": 15,
      "position": "absolute",
      "right": 0,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#ff323c",
        }
      }
    >
      Test Text
    </Text>
  </View>
</View>
`;
